starknet_py/contract.py

```diff
@@ -224,10 +225,16 @@ async def invoke(
 
         return invoke_result
 
-    async def estimate_fee(self):
+    async def estimate_fee(
+        self,
+        block_hash: Optional[Union[Hash, Tag]] = None,
+        block_number: Optional[Union[int, Tag]] = None,
+    ):
         """
         Estimate fee for prepared function call
 
+        :param block_hash: Estimate fee at specific block hash
+        :param block_number: Estimate fee at given block number (or "pending" for pending block), default is "pending"
```

Comment:
Is it though? It doesn't seem like it from the function declaration?

Reply:
All methods which use block_hash or block_number and FullNode/Gateway Client has the block_number parameter default to "pending"

Reply:
Just to be safe in case we change anything in Client I think we could also include default parameter in `Contract` anyways.